engine/src/main/scala/cromwell/engine/workflow/tokens/TokenQueue.scala

```diff
@@ -47,7 +48,14 @@ final case class TokenQueue(queues: Map[String, Queue[TokenQueuePlaceholder]],
     * Returns a dequeue'd actor if one exists and there's a token available for it
     * Returns an updated token queue based on this request (successful queuees get removed, hogs get sent to the back)
     */
-  def dequeue: DequeueResult = recursingDequeue(queues, Vector.empty, queueOrder)
+  def dequeue: DequeueResult = {
+    val guaranteedNonEmptyQueues = queues.filterNot { case (hogGroup, q: Queue[_]) =>
+      val empty = q.isEmpty
+      if (empty) logger.warn(s"Programmer error: Empty token queue value still present in TokenQueue: $hogGroup")
+      empty
+    }
+    recursingDequeue(guaranteedNonEmptyQueues, Vector.empty, queueOrder)
+  }
 
   private def recursingDequeue(queues: Map[String, Queue[TokenQueuePlaceholder]], queuesTried: Vector[String], queuesRemaining: Vector[String]): DequeueResult = {
     if (queuesRemaining.isEmpty) {
```

Comment:
[ToL] I wonder if we could put this empty check in `dequeue` and skip calling `recursingDequeue` at all

Just a way of putting two related checks - list not empty & no empty items in list - in the same place

Reply:
I think it could be in `dequeue` as well - but this looks like it serves as the base case escape for the recursion function so I think it's still needed here as well